src/classes/Advancement_Info.cls

```diff
@@ -70,35 +70,45 @@ public with sharing class Advancement_Info {
         }
 
-        //Check if Advancement is installed
-        try {
-            Schema.DescribeSObjectResult[] descResult = Schema.describeSObjects(new String[]{hs.Advancement_Namespace__c + '__Advancement_Setting__mdt'});
-            isManaged = true;
-        } catch (Exception e) {
-            try {
-                Schema.DescribeSObjectResult[] descResult = Schema.describeSObjects(new String[]{'Advancement_Setting__mdt'});
-            } catch (Exception ex) {
-                setupCalled = true;
-                return;
+        //Check if GEM is installed
+        List<String> settingsObjs = new List<String> {'Advancement_Setting__mdt', 'GEM_API_Settings__mdt'};
+        List<String> useGEMFields = new List<String> {'Use_Advancement__c', 'Use_GEM__c'};
+
+        boolean found = false;
+
+        Integer i=0;
+        while (!found && i < settingsObjs.size()) {
+            try { // With Namespace
+                Schema.DescribeSObjectResult[] descResult = Schema.describeSObjects(new String[]{hs.Advancement_Namespace__c + '__' + settingsObjs[i]});
+                isManaged = true;
+                found = true;
+            } catch (Exception e) {
+                try { // Without Namespace
+                    Schema.DescribeSObjectResult[] descResult = Schema.describeSObjects(new String[]{settingsObjs[i]});
+                    found = true;
+                } catch (Exception ex) {
+                    i++;
+                }
             }
         }
 
+        if (!found) {
+            setupCalled = true;
+            return;
+        }
+
+        String gemAPISettingsObj = settingsObjs[i];
+        String useGEMField= useGEMFields[i];
 
         //If Adv is installed, we need to throw exception if Adv is not configured correctly
         Boolean hasConfigError = true;
 
-        String advSettingQuery = 'select Use_Advancement__c, API_HEDA_Class__c from Advancement_Setting__mdt';
-        String useAdvStr = 'Use_Advancement__c';
-        String apiHedaClassStr = 'API_HEDA_Class__c';
+        String nsPrefixOrBlank = (isManaged) ? hs.Advancement_Namespace__c + '__' : '';
 
-        if (isManaged == true) {
-            advSettingQuery = 'select ' + hs.Advancement_Namespace__c + '__' + 'Use_Advancement__c, ' +
-                    hs.Advancement_Namespace__c + '__' + 'API_HEDA_Class__c from ' +
-                    hs.Advancement_Namespace__c + '__' + 'Advancement_Setting__mdt';
-            useAdvStr = hs.Advancement_Namespace__c + '__' + 'Use_Advancement__c';
-            apiHedaClassStr = hs.Advancement_Namespace__c + '__' + 'API_HEDA_Class__c';
+        String useAdvStr = nsPrefixOrBlank + useGEMField;
+        String apiHedaClassStr = nsPrefixOrBlank + 'API_HEDA_Class__c';
 
-        }
+        String advSettingQuery = 'SELECT ' + useAdvStr + ', ' + apiHedaClassStr + ' FROM ' + nsPrefixOrBlank + gemAPISettingsObj;
```

Comment:
@j-brandolini Hey John & @lparrott -- do we need to worry about data cleansing here? Stripslashes, etc?

Reply:
@lifewithryan - I am not sure what you mean.  Would you mind giving a specific example?  Sorry if I am being dense!

Reply:
@lifewithryan We should be ok here, none of the query parts actually come from text entered by the user, we're just checking for _which_ object name is currently being used in the org, so we know all of the possibilities.

Reply:
@lifewithryan - I am with you now...you are worried about SOQL Injection.  The only part that would even potentially be an issue is the namespacePrefix.  It is not protected, but only an admin could update it.  If you are worried about it, I will escape the prefix.  Let me know...

Reply:
best practice would be to do so. I'm sure it would NEVER happen but a disgruntled admin, or someone with elevated access (we've all see the org where everyone's an admin) could mess some stuff up. PLUS -- not sure if we ever have to go through security review again, but something like that might bite us...perhaps?

Reply:
I will add it.  It is easy enough.

Reply:
Agree! Pretty sure it would get flag in a sec-review recert.

Reply:
I will make sure to do it in the HEDA and NPSP pull requests on Monday.

Reply:
@j-brandolini just let me know when it is done.

Reply:
After doing a triple check, the only field that is coming in that needed to be escaped was the namespace field.  I have updated for this.